sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statements.scala

```diff
@@ -51,6 +51,14 @@ abstract class ParsedStatement extends LogicalPlan {
   final override lazy val resolved = false
 }
 
+abstract class ParsedTableStatement extends ParsedStatement {
```

Comment:
@cloud-fan, I added these changes to fix view error messages.

There were 2 failing tests, but I realized while fixing them that temporary tables were getting converted to v2 plans. I can fix the failing tests and separate out the other changes to a new PR if you'd prefer.

The new changes avoid converting temporary views in `ResolveCatalogs` and in most rules in `ResolveSessionCatalog` by updating the `NonSessionCatalog` and `SessionCatalog` extractors to ignore identifiers that are temporary views. Then, I updated `CheckAnalysis` to throw better exceptions using these new superclasses, `ParsedTableStatement` and `ParsedViewStatement`.

I also had to update a few rules in `ResolveSessionCatalog` that are used to create v1 plans for views. Oddly, `DropTableCommand` is used to drop views in several tests.

Reply:
+1 to add them here. This is the right way to fix the error message.